packages/maps/src/components/MapView.tsx

```diff
@@ -17,13 +19,42 @@ function zoomToAltitude(zoom: number) {
   return C * Math.pow((A - D) / (zoom - D) - 1, 1 / B);
 }
 
-class MapView extends React.Component<MapViewProps> {
+type State = {
+  userLocation?: {
+    latitude: number;
+    longitude: number;
+    altitude: number | null;
+    accuracy: number | null;
+    altitudeAccuracy: number | null;
+    heading: number | null;
+    speed: number | null;
+  };
+};
+
+class MapView extends React.Component<MapViewProps, State> {
   private mapRef: React.RefObject<any>;
   constructor(props: MapViewProps) {
     super(props);
+    this.state = {};
     this.mapRef = React.createRef();
   }
 
+  componentDidMount() {
+    (async () => {
+      if (!this.props.showUserLocation) {
+        return;
+      }
+
+      let { status } = await Location.requestForegroundPermissionsAsync();
+      if (status !== "granted") {
+        return;
+      }
+
+      let location = await Location.getCurrentPositionAsync({});
```

Comment:
I believe these can both be `const` because they're not reassigned

```suggestion
      const { status } = await Location.requestForegroundPermissionsAsync();
      if (status !== "granted") {
        return;
      }

      const location = await Location.getCurrentPositionAsync({});
```

Reply:
Good call!